### PR TITLE
Allow creating and restoring backups from C

### PR DESCRIPTION
**Pull request by mkevac**
Added simple wrappings allowing backing up RocksDB in C.
I don't have a lot of experience in C++, so comments are welcome.

### CODE REVIEW

**Comment by mkevac** [comment]
Any updates?

File: db/c.cc [hunk 1]
```diff
@@ -527,6 +531,32 @@ rocksdb_t* rocksdb_open_for_read_only(
   return result;
 }
 
+rocksdb_backup_engine_t* rocksdb_backup_engine_open(
+    const char* path,
+    char** errptr) {
+  BackupEngine* be;
+  if (SaveError(errptr, BackupEngine::Open(Env::Default(), BackupableDBOptions(path), &be))) {
```
Comment (igorcanadi) [inline db/c.cc]: Can we also make Env* configurable? Clients might want to pass HdfsEnv here, effectively backing up RocksDB to HDFS.

**Comment by igorcanadi** [comment]
Thanks for contributing! Can you also please add a unit test for the new feature? Also, please run `make format` to fix the styling.

**Comment by mkevac** [comment]
* Added unit tests
* Added some more wrappers
* Made an ENV configurable

I have tried `make format`, but it make a code look nothing like a code around. Are you sure it works as it should?

**Comment by igorcanadi** [comment]
Looks great! Don't worry about `make format`, I'll just run it on your commit once we merge it.

Just couple of nits:
* Can we also add restore procedure to c_test? Since we're backing up, we might as well show how to restore :)
* Can you please add comment on c_simple_example? I know code is mostly self-explanatory, but I think it would help some people.

Tnx!

**Comment by mkevac** [comment]
I have already added restoration check/procedure to c_test. Have you missed it? Or maybe you are talking about adding it to c_simple_example?

**Comment by igorcanadi** [comment]
Yeah, I was thinking about adding it to c_simple_example. The test looks good!

**Comment by mkevac** [comment]
Done.

**Comment by igorcanadi** [comment]
Great stuff!

**Comment by mkevac** [comment]
Thank you.

**Pull request merged**